src/SDK/SmartDetectorsShared/Clients/TelemetryDataClientBase.cs

```diff
@@ -177,7 +177,12 @@ public async Task<IList<DataTable>> RunQueryAsync(string query, TimeSpan? dataTi
                     }
 
                     // Try again with filtered resources
-                    response = await this.InternalRunQueryAsync(query, filteredAdditionalTelemetryResourceIds, telemetryId, dataTimeSpan, cancellationToken);
+                    response = await this.InternalRunQueryAsync(query, filteredAdditionalTelemetryResourceIds, telemetryId, dataTimeSpan, this.queryUriFormat, cancellationToken);
+                }
+                else if (response.StatusCode == HttpStatusCode.NotFound)
+                {
+                    // Try again to AIMON
+                    response = await this.InternalRunQueryAsync(query, iterationTelemetryResourcesIds, telemetryId, dataTimeSpan, "https://api.aimon.applicationinsights.io/v1/apps/{0}/query", cancellationToken);
```

Comment:
what's all of this code?

Reply:
mu bad

---
In reply to: [242884990](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/36#discussion_r242884990) [](ancestors = 242884990)